include/sdsl/divsufsort.hpp

```diff
@@ -0,0 +1,1973 @@
+/*
+ * libdivsufsort
+ * Copyright (c) 2003-2008 Yuta Mori All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef INCLUDED_SDSL_DIVSUFSORT
+#define INCLUDED_SDSL_DIVSUFSORT
+
+namespace sdsl {
+
+#define PROJECT_VERSION_FULL "2.0.1-15-g22e6b23"
+
+#include <assert.h>
+#include <stdio.h>
+#include <inttypes.h> // alternative: #include <stdint.h>
+
+#ifdef _OPENMP
+# include <omp.h>
+#endif
+
+// #include <stdlib.h>
+// #include <memory.h>
+// #include <stddef.h>
+// #include <strings.h>
+
+#if !defined(UINT8_MAX)
+# define UINT8_MAX (255)
+#endif
+
+#define ALPHABET_SIZE (256)
+#define BUCKET_A_SIZE (ALPHABET_SIZE)
+#define BUCKET_B_SIZE (ALPHABET_SIZE * ALPHABET_SIZE)
+#define SS_INSERTIONSORT_THRESHOLD (8)
+#define SS_BLOCKSIZE (1024)
+#define SS_MISORT_STACKSIZE (16)
+#define TR_INSERTIONSORT_THRESHOLD (8)
```

Comment:
why are these not part of the config?

Reply:
They don't depend on 32/64 bit versions. We could move them in a separate config struct. But I only want to change the code when necessary.